### DISCUSSION TITLE
Device Token Registration Failed (iOS)

### DISCUSSION THREAD

**Discussion by HanweeeeLee** [WalletConnect/walletconnect-monorepo#3938] [Developer Support]
Hello.
I am a developer who is developing Wallet using WalletConnect.

Thankfully, I am using the service and SDK you developed.

This time, we are linking Wallet Connect to our service, and when we send a signature request from Dapp (PC's web browser) to Wallet, we add a Notify package to send a push to our mobile device.

Based on what you wrote in the guide
The following code has been added to our code.
```
func application(
      _ application: UIApplication,
      didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
) {
    Task(priority: .high) {
        try await Notify.instance.register(deviceToken: deviceToken)
    }
}
```

I expected there would be no problem because it was simply a code to register a device token, but I'm getting 400 errors back.

Looking at the SDK's code, it seems that if you call the 'register' method of PushAPI HTTP communication, I'm getting a 400 error return.

The parameters look like the following.
```
[
    "client_id": clientId,
    "type": environment.rawValue,
    "token": token
]
```
When I printed the log on LLDB, my client ID is "z6MkeyXbzx4u9ZiBwNpCdtWktk5dKNndwW9ss2JUtumDYo66".

FCM information has also been entered in the developer console.

Is there anything else I can look at?

Thank you for always helping me.

**Comment by chris13524** [comment]
Are you using the Swift SDK or are you making a custom implementation? What is the response body of the 400 error? In addition to the request body you also need to supply an Authorization header.

**Reply by HanweeeeLee** [reply]
Hello.
Thank you for your response.

First of all, I am using the SDK provided by WalletConnect, and I checked that if I use the register method provided by the SDK, the authentication key is also in the header.

I did some analysis on this problem.
1. I checked that Android Push is working normally.
2. Eventually, HTTP communication (/\(projectId)/clients) will cause 400 errors, so I compared HTTP related information such as Body (between iOS and Android)
3. As a result of checking, it was recognized that the type value of the body should be put as "fcm".
4. As a result of looking into the SDK code, I recognized that the current SDK code cannot contain the value "fcm". (I might not have found it. I will attach an [enum value](https://github.com/search?q=repo%3AWalletConnect%2FWalletConnectSwiftV2%20apns-sandbox&type=code) that can be entered as the value of the type below.)
```
public enum APNSEnvironment: String {
    case production = "apns"
    case sandbox = "apns-sandbox"
}
```
5. After some code modifications, I added the "fcm" value, and I checked that the push was working. (This is a temporary modification of the code so that only "fcm" values can be entered, not perfect modification of the code.)

Thank you for paying attention to my problem.

**Reply by chris13524** [reply]
On iOS you need setup your APNS credentials in Cloud instead of FCM. I do not believe you can use FCM on iOS as that is for Android.

**Reply by HanweeeeLee** [reply]
Thanks for the reply. I recognized the problem, and I recognized which direction to fix it.
